### ISSUE TITLE
No hints due to semicolon

### ISSUE THREAD

**Issue by DIMANRUS** [dbcodeio/public#46]
Hello, If I don't put a semicolon, there won't be any hints for slelect below
![image](https://github.com/user-attachments/assets/83c0be03-1ee7-42da-acc3-5dc97ae0443a)
But with a semicolon they are
![image](https://github.com/user-attachments/assets/f6aef428-e2eb-4b9e-a568-82cb0e897b93)

**Comment by mikeburgh** [comment]
Yes, the parser is using the semi colon as the statement boundary.. so it knows where one starts and ends.

I might be able to change that to look for select/insert/update/delete/create/etc... but that's going to lead to other issues when I get around to it being able to understand CTE's

**Comment by DIMANRUS** [comment]
Okay. I'll put a semicolon then.

**Comment by mikeburgh** [comment]
Version 1.4.8 is now out, which has support for not having semi colons between statements.<br><br>If statements above contain invalid syntax, that may cause autocomplete to stop working, as I get a bit closer with the parser, I will be able to indicate those syntax errors and highlight them.

Let me know how you go with it.

**Comment by DIMANRUS** [comment]
> Version 1.4.8 is now out, which has support for not having semi colons between statements.<br><br>If statements above contain invalid syntax, that may cause autocomplete to stop working, as I get a bit closer with the parser, I will be able to indicate those syntax errors and highlight them.
> 
> Let me know how you go with it. 

It would be great to highlight the error. Recently there have been cases where autocompletion stopped working.

**Comment by mikeburgh** [comment]
I just pushed version 1.4.10 which has the fallback autocomplete in it.. which should smooth over cases where the parser is not able to figure out the context. 

Once the parser has more of the syntax sorted, I will add the highlighting, just need to get it to a point where it has 90% or more of the SQL syntax in place..

**Comment by mikeburgh** [comment]
Closing this for now, highlighting is a way off still, and the fallback seems to have caught a lot of edge cases
